internal/types/init.go

```diff
@@ -1,22 +1,35 @@
 package types
 
-// InitialComposeFile TODO
+const defaultComposefileVersion = "3.6"
+
+// InitialComposeFile represents an initial composefile (used by the init command)
 type InitialComposeFile struct {
 	Version  string
-	Services *map[string]InitialService
+	Services map[string]InitialService
 }
 
-// InitialService TODO
+// InitialService represents an initial service (used by the init command)
 type InitialService struct {
 	Image string
 }
 
 // NewInitialComposeFile returns an empty InitialComposeFile object
-func NewInitialComposeFile() InitialComposeFile {
-	services := make(map[string]InitialService)
+func NewInitialComposeFile(ops ...func(*InitialComposeFile)) InitialComposeFile {
```

Comment:
Isn't this `ops` parameter only used in test code? I'm not fan of "test" related code in production code, it acts as dead code for me.

Reply:
Right… so.. it's mainly preparing for being able to `initialize` a composefile with other versions or services but it could definitely go away :stuck_out_tongue_closed_eyes: